src/table/boot.rs

```diff
@@ -405,24 +413,28 @@ impl BootServices {
         watchdog_code: u64,
         data: Option<&mut [u16]>,
     ) -> Result<()> {
+        assert!(
+            watchdog_code > 0xffff,
+            "Invalid use of a reserved firmware watchdog code"
+        );
```

Comment:
Cross-checked the spec, and I'm not sure if these reserved codes can even be used anymore. Better safe than sorry.